src/HtmlReader/index.tsx

```diff
@@ -20,7 +21,8 @@ export type HtmlAction =
   | { type: 'SET_COLOR_MODE'; mode: ColorMode }
   | { type: 'SET_SCROLL'; isScrolling: boolean }
   | { type: 'SET_FONT_SIZE'; size: number }
-  | { type: 'SET_FONT_FAMILY'; family: FontFamily };
+  | { type: 'SET_FONT_FAMILY'; family: FontFamily }
+  | { type: 'SET_CURRENT_TOC_URL'; currentTocUrl: string };
```

Comment:
I see, I do also need to update the goForward and goNext function, I was wondering if it's good to expose the D2 object to the component, which should contain the updated positions.

Reply:
What do you mean expose the D2 object to the component?

Reply:
Pass down the D2(reader) object to the TOC component so we can get the position directly instead of adding the extra reducer.

Reply:
We shouldn't pass down the D2Reader because the UI components shouldn't know about it. We should just pass down the `currentLocation`, so in general I think this is ok